### PR TITLE
fix(persist): fix async migrate on persist middleware

### PR DESCRIPTION
**Pull request by dbritto-dev**
## Related Bug Reports or Discussions

Fixes #2827

## Check List

- [x] `pnpm run prettier` for formatting code and docs

### CODE REVIEW

**Comment by KiwiKilian** [comment]
I tried to fix it way to complicated, looking good, thank you!

File: .github/workflows/test-multiple-builds.yml [hunk 1]
```diff
@@ -37,11 +37,11 @@ jobs:
       - name: Patch for CJS
         if: ${{ matrix.build == 'cjs' }}
         run: |
-          sed -i~ "s/resolve('\.\/src\(.*\)\.ts')/resolve('\.\/dist\1.js')/" vitest.config.ts
+          sed -i~ "s/resolve('\.\/src\(.*\)\.ts')/resolve('\.\/dist\1.js')/" vitest.config.mts
```
Comment (dai-shi) [inline .github/workflows/test-multiple-builds.yml:40]: Is this change necessary for this PR?
Can you open a separate PR and open the same PRs to `jotai` and `valtio` too?

**Review by dai-shi — COMMENTED** [review]
Looks good to me otherwise! Thanks for your work!

File: .github/workflows/test-multiple-builds.yml (hunk 1 above)
Comment (dbritto-dev) [reply .github/workflows/test-multiple-builds.yml:40]: @dai-shi sure, let me create a new PR for that

**Comment by dbritto-dev** [comment]
@dai-shi all yours

**Review by dai-shi — APPROVED** [review]
Thanks!

**Comment by dbritto-dev** [comment]
@dai-shi jobs are stuck
